lms/djangoapps/email_marketing/tasks.py

```diff
@@ -294,6 +296,72 @@ def _get_course_content(course_url, sailthru_client, email_config):
     return response
 
 
+def _get_or_create_user_lists(sailthru_client, site_domain=None):
+    """
+    Get the user list name from cache if exists else create one and return the name
+    :param sailthru_client:
+    :param user_lists:
+    :return: list name if exists or created else return None
+    """
+    email_config = EmailMarketingConfiguration.current()
+    if not email_config.enabled:
+        return
+
+    if site_domain and "edx.org" not in site_domain:
+        list_name = site_domain.replace(".", "_") + "_user_list"
+    else:
+        list_name = "All edX Users"
+
+    cached = cache.get(email_config.CACHE_KEY)
+    if cached:
+        for user_list in cached['lists']:
+            if user_list.get('name') == list_name:
+                return list_name
+
+    sailthru_response = _get_or_create_list(sailthru_client, list_name)
+    if sailthru_response:
+        cache.set(email_config.CACHE_KEY, sailthru_response)
+        return list_name
```

Comment:
Move cache population to `_get_or_create_sailthru_list`.
Replace this block with:
```
_get_or_create_sailthru_list(sailthru_client, list_name)
return list_name
```